compaction/compaction_manager.cc

```diff
@@ -1035,6 +1035,13 @@ void compaction_manager::register_metrics() {
 
 void compaction_manager::enable() {
     SCYLLA_ASSERT(_state == state::none || _state == state::disabled);
+
+    --_disabled_state_count;
```

Comment:
Idea: change the `enabled` / `disabled` state to use an auxiliary counter
and leave only `none|running|stopped` in `compaction_manager::state` enum class.
And then disabled enabled state would be valid only in the `running` state and be determined by a counter.

Reply:
Good idea. I will check if `stopped` is used only after a service is stopped, which should according to the documentation. Then we would get rid of all these states as the service cannot be restarted once stopped.

Reply:
Done.